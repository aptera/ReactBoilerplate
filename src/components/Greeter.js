// This is the preferred coding style (ES6 native)

import React, {Component} from 'react';
import Button from './Button';

class Greeter extends Component {
    constructor(){
        super();
        this.state = {
            currentNumber: 0
        }

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        var next = this.state.currentNumber + 1;
        this.setState({ currentNumber: next });
    }

    render(){
        return(
            <div>
                <div>I've Said Hello {this.state.currentNumber} Times.</div>
                <Button click={this.handleClick} text="Say Hello Biotch" />
            </div>
        );
    }
}

export default Greeter;