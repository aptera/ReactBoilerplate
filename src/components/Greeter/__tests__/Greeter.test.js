import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

// import the component to put under test.
import Greeter from '../Greeter';

describe('the progress bar component', () => {

    let GreeterComponent;
    beforeEach(() => {
        GreeterComponent = ReactTestUtils.renderIntoDocument(<Greeter />);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <Greeter />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('will emphatically begin a sentence based on the number of times it has been required to say hello', () => {
        expect(GreeterComponent.getEmphaticPre()).toBe('Hello, ');
        GreeterComponent.state.currentNumber = 5;
        expect(GreeterComponent.getEmphaticPre()).toBe('Hey, ');
        GreeterComponent.state.currentNumber = 10;
        expect(GreeterComponent.getEmphaticPre()).toBe('Dang it!, ');
    });

    it('will emphatically end a sentence based on the number of times it has been required to say hello', () => {
        expect(GreeterComponent.getEmphaticPost()).toBe('.');
        GreeterComponent.state.currentNumber = 5;
        expect(GreeterComponent.getEmphaticPost()).toBe('!');
        GreeterComponent.state.currentNumber = 10;
        expect(GreeterComponent.getEmphaticPost()).toBe('!!!!!');
    });

    it('will increment the count when the button is clicked', () => {
        expect(GreeterComponent.state.currentNumber).toBe(0);
        GreeterComponent.handleClick();
        expect(GreeterComponent.state.currentNumber).toBe(1);
    });

});