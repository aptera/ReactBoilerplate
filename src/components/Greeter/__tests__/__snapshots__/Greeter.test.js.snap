// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the progress bar component renders correctly 1`] = `
<div
  className="greeter"
>
  <div>
    Hello, 
    I've said hello 
    0
     times
    .
  </div>
  <div>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Say Hello
    </button>
  </div>
</div>
`;
